api/queries/user/githubProfile.js

```diff
@@ -1,10 +1,50 @@
 // @flow
+require('now-env');
 import type { DBUser } from 'shared/types';
+import cache from 'shared/cache/redis';
+import fetch from 'isomorphic-fetch';
 
-export default ({ githubProviderId, githubUsername }: DBUser) => {
-  if (!githubProviderId || !githubUsername) return null;
-  return {
-    id: githubProviderId,
-    username: githubUsername,
-  };
+export default async ({ githubProviderId, id }: DBUser) => {
+  if (!githubProviderId) return null;
+
+  const cachedGithubProfile = await cache.get(`githubProfile:${id}`);
+
+  if (cachedGithubProfile) {
+    const parsed = JSON.parse(cachedGithubProfile);
+    return {
+      id: parsed.id,
+      username: parsed.username,
+    };
+  } else {
+    const apiUrl = 'https://api.github.com';
+    const apiRoute = '/user/';
+    const { GITHUB_OAUTH_CLIENT_ID, GITHUB_OAUTH_CLIENT_SECRET } = process.env;
+    const queryString = `?client_id=${GITHUB_OAUTH_CLIENT_ID}&client_secret=${GITHUB_OAUTH_CLIENT_SECRET}`;
```

Comment:
nit: let's move the static variables outside of the resolver to the top of the file